File: PrefixSum/bookingFlight.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        int bookSize = bookings.size();
        vector<int>ans(n,0);
        for(int i=0;i<bookSize;i++){
            int startIdx = bookings[i][0]-1;
            int endIdx = bookings[i][1]-1;
            for(int j=startIdx;j<=endIdx;j++){
                ans[j] += bookings[i][2];
            }
        }
        return ans;
    }
};

File: PrefixSum/bookingFlight.py
from typing import List

class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        bookSize = len(bookings)
        ans = [0]*n
        for i in range(bookSize):
            startIdx = bookings[i][0]-1
            endIdx = bookings[i][1]-1
            for j in range(startIdx,endIdx+1):
                ans[j] += bookings[i][2]
        return ans

